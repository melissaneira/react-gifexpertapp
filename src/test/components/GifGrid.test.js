import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid";
import '@testing-library/jest-dom';
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Probando componente GifGrid', () => { 
    const category = 'One Piece';
    test('Debe renderizar correctamente el componente', () => {         
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
     })

    test('debe de mostrar items cuando se carga las imagenes useFetchGifs', () => { 
        const gifs = [{
            id: 'ABC',
            url:'https://localhost/file/figure.jpg',
            title: 'Figure1'
        },
        {
            id: 'DEF',
            url:'https://localhost/file/figure.jpg',
            title: 'Figure2'
        },
        {
            id: 'GHI',
            url:'https://localhost/file/figure.jpg',
            title: 'Figure3'
        }
    ]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category}/>);        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
     })
 })