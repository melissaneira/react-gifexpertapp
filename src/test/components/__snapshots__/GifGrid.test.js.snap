// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente GifGrid Debe renderizar correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Piece
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Probando componente GifGrid debe de mostrar items cuando se carga las imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One Piece
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Figure1"
      url="https://localhost/file/figure.jpg"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="Figure2"
      url="https://localhost/file/figure.jpg"
    />
    <GifGridItem
      id="GHI"
      key="GHI"
      title="Figure3"
      url="https://localhost/file/figure.jpg"
    />
  </div>
</Fragment>
`;
